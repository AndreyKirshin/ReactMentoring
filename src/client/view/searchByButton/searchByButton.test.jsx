
import React from 'react';
import { shallow, mount } from 'enzyme';
import SearchByButton from './searchByButton';

describe('SearchByButton component', () => {

    let name;
    let toggle;
    let isActive;
    
    beforeAll(() => {
        name = 'TITLE';
        toggle = jest.fn();
        isActive = true;
    });

    it('should be render correctly', () => {
        const component = shallow(
            <SearchByButton 
                name={name} 
                toggle={toggle} 
                isActive={isActive} 
            />);
        expect(component).toMatchSnapshot();
    });

    it('should be render with class search-by-active if isActive is true', () => {
        const component = shallow(
            <SearchByButton 
                name={name} 
                toggle={toggle} 
                isActive={isActive} 
            />);
        expect(component.hasClass('search-by-active')).toBe(true);
    });
})