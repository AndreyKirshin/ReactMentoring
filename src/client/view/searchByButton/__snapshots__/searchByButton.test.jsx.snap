// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchByButton component should be render correctly 1`] = `ShallowWrapper {}`;
